tools/model_tools/src/openvino/model_zoo/open_model_zoo.py

```diff
@@ -0,0 +1,204 @@
+
+        flags = ['--name=' + model_name,
+                 '--output_dir=' + str(download_dir),
+                 '--cache_dir=' + str(cache_dir),
+                 '--precisions=' + str(precision)]
+
+        omz_downloader.download(flags)
+
+        description = '{}\n\n    License: {}'.format(model.description, model.license_url)
+        task_type = model.task_type
+
+        if precision not in model.precisions:
+            raise ValueError('Incorrect precision value!\n    Current precision value is: \''
+                                        + str(precision) + '\'.\n    Allowed precision values for model are: '
+                                        + str(model.precisions))
+
+        prefix = Path(model_dir) / precision / model_name
+
+        model_path = str(prefix) + '.xml'
+        bin_path = str(prefix) + '.bin'
+        name = model.name
+        subdirectory = model.subdirectory
+
+        if not os.path.exists(model_path) or not os.path.exists(bin_path):
+            omz_converter.converter(['--name=' + model_name, '--precisions=' + precision,
+                        '--download_dir=' + str(download_dir)])
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    @classmethod
+    def from_pretrained(cls, model_path, *, task_type=None, ie=None):
+        '''
+        Loads model from existing .xml, .onnx files.
+        Parameters
+        ----------
+            model_path
+                Path to .xml or .onnx file.
+            task_type
+                Type of task that the model performs.
+            ie
+                Inference Engine instance
+        '''
+        model_path = Path(model_path).resolve()
+
+        if not model_path.exists():
+            raise ValueError('Path {} to model file does not exist.'.format(model_path))
+
+        if model_path.suffix not in ['.xml', '.onnx']:
+            raise ValueError('Unsupported model format {}. Only .xml or .onnx supported.'.format(model_path.suffix))
+
+        description = 'Pretrained model {}'.format(model_path.name)
+        name = model_path.name
+        model = cls._load_model(cls, name)
+        task_type = task_type
+        subdirectory = model.subdirectory if model else model_path.parent
+        model_path = str(model_path)
+
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    def _load_model(self, model_name):
+        parser = argparse.ArgumentParser()
+        args = argparse.Namespace(all=False, list=None, name=model_name, print_all=False)
+        try:
+            model = _configuration.load_models_from_args(parser, args, _common.MODEL_ROOT,
+                        mode=_configuration.ModelLoadingMode.ignore_composite)[0]
+        except SystemExit:
+            model = None
+
+        return model
+
+    def __call__(self, inputs, device='CPU'):
+        if self.ie is None:
+            raise TypeError('ie must be of IECore type.')
```

Comment:
message is not clear.
There are 2 possible situations:
1. self.ie is None - ie is not specified (during creating model for example
2. self.ie is provided, but it is object of wrong type (e.g. instead of openvino.inference_engine.IECore provided openvino.runtime.Core or something different)

so your exception looks misleading, if user does not specify ie at all

Reply:
BTW, do you have plans to migrate on OV2.0? (it has some differences)

Reply:
Messages updated

> BTW, do you have plans to migrate on OV2.0? (it has some differences)

I do. It is in progress